pkg/sql/catalog/descpb/structured.proto

```diff
@@ -1073,6 +1073,12 @@ message DatabaseDescriptor {
   // name resolution to know without a KV lookup whether a database has a
   // child schema with a target name. Temporary schemas are not stored here.
   map<string, SchemaInfo> schemas = 7 [(gogoproto.nullable) = false];
+
+  enum State {
```

Comment:
[nit]: comment

It feels bad to define one of these states for every kind of descriptor, but not sure if there is a better way

Reply:
We could move the table descriptor's `State` enum to the top level and use that everywhere, but I think we'd want to decide to do that now, before the beta gets cut.

I think it's good to have the safety of not being able to put a database in the adding state, and keeping the different definitions at the level of the wrappers, etc. doesn't feel too bad. But I don't have a strong opinion either way.

Reply:
yeah i think its fine so that we have the flexibility to have different states for each descriptor type. It just feels a bit odd though.